src/ngraph/runtime/backend.cpp

```diff
@@ -62,16 +51,24 @@ void* runtime::Backend::open_shared_library(string type)
     {
         type = type.substr(0, colon);
     }
-    string lib_name = "lib" + to_lower(type) + "_backend" + ext;
+
+    vector<string> backend_list;
+    string version = NGRAPH_VERSION;
+    version = regex_replace(version, regex("\\."), "\\.");
+    version = regex_replace(version, regex("\\+"), "\\+");
+    regex reg("lib" + to_lower(type) + "_backend.*" + version);
```

Comment:
Maybe stick a `^` at the beginning and a `\.so$` (or `\.dylib$` for Mac, which is a pain) at the end?

Reply:
Done. I had defined a compiler def SHARED_LIB_EXT because cmake knows the correct extension.